packages/core/rfcs/composable-relations.md

```diff
@@ -0,0 +1,86 @@
+# Composable relations
+
+## Motivation
+
+By injecting composables to the others we can achieve an open to extensions dependency management.
+
+## Case study
+
+We have two composables:
+
+- useUser
+- useUserAddress
+
+We use GraphQL CommerceTools and in this case editting user's addresses is through the user collection. This approach require update the user object after each modifications in their adresses. That means we need access to user entity through useUserAddress composable
+
+
+### Current approach
+
+In order to get access to the user through the useUserAddress we need to following changes (please treat this more as a pseudo code, rather than a strict one):
+
+```TS
+// useUser composable
+export const user = ref(null);
+```
+
+```TS
+// useUserAddress composable
+import { user } from 'useUserComposable';
+
+export const params = {
+  addAddress: (address) => {
+    const addresses = await addNewAddressGraphQL(address);
+
+    user.value = { ...user.value, addresses };
+
+    return addresses;
+  }
+};
+```
+
+As we see, this approach has some overcomplexity solutions:
+- mutations in factories and composables as well
+- the user object is now public and anyone can avoid the architecture principles
+- too much logic in composables. The solution is not agnostic anymore
+
+### Proposed approach
+
+Instead of exporting the user object, we can inject an optional (or not) useUser composable through factory params. We can also define logic depending on is useUser provided or not.
+
+Examples:
+
+```TS
+//  useUser composable
+
+const user = ref(null); // user object is now a private property
+```
+
+```TS
+// useUserAddressFactory
+
+interface UseUserAddressFactoryParams {
+  userComposable?: UseUser<USER, USER_UPDATE_PARAMS>;
+  addAddress: { userUpdated: USER, addressesUpdated: ADDRESS[] }
+}
+
+export function useUserAddressFactory {
```

Comment:
that's the weird example it's not even a factory but looks like factory params

Reply:
I was thinking about this DI etc. and it ONLY make sense when you need to call other composable INSIDE of the factory, but this way is a little bit risky as some of these calls can depend on the integration. Using factory params you can do something like this

```
const someComposable = useSomeComposable();

const factoryParams = {
  someFunction: async () => {
    const field = someComposable.someField;
  }
}
```

Reply:
@defudef so please provide other examples where it won't work and you need DI mechanism

Reply:
I'm actually against calling it that way tbh, we should have a way of performing such stuff from the inside of the factory so the control is on our side, O believe this is the only reason why we need such thing. Ideally we should have some kind of internal linking between compostables that needs to be linked (i believe it applies only to user

Reply:
Sorry, just missed brackets:

```TS
export function useUserAddressFactory() {
 // that stuff inside here
}
````